src/java.base/share/classes/jdk/internal/util/HexDigits.java

```diff
@@ -103,20 +92,24 @@ private HexDigits() {
     }
 
     /**
-     * Combine two hex shorts into one int based on big endian
+     * Return a little-endian packed integer for the 4 ASCII bytes for an input unsigned 2-byte integer.
+     * {@code b0} is the most significant byte and {@code b1} is the least significant byte.
+     * The integer is passed byte-wise to allow reordering of execution.
      */
-    public static int digit(int b0, int b1) {
-        return (DIGITS[b0 & 0xff] << 16) | DIGITS[b1 & 0xff];
+    public static int packDigits(int b0, int b1) {
+        return DIGITS[b0 & 0xff] | (DIGITS[b1 & 0xff] << 16);
     }
 
     /**
-     * Combine four hex shorts into one long based on big endian
+     * Return a little-endian packed long for the 8 ASCII bytes for an input unsigned 4-byte integer.
+     * {@code b0} is the most significant byte and {@code b3} is the least significant byte.
+     * The integer is passed byte-wise to allow reordering of execution.
      */
-    public static long digit(int b0, int b1, int b2, int b3) {
-        return (((long) DIGITS[b0 & 0xff]) << 48)
-                | (((long) DIGITS[b1 & 0xff]) << 32)
-                | (DIGITS[b2 & 0xff] << 16)
-                | DIGITS[b3 & 0xff];
+    public static long packDigits(int b0, int b1, int b2, int b3) {
+        return DIGITS[b0 & 0xff]
+                | (DIGITS[b1 & 0xff] << 16)
+                | (((long) DIGITS[b2 & 0xff]) << 32)
+                | (((long) DIGITS[b3 & 0xff]) << 48);
```

Comment:
Can you reverse the order of these source lines to put the shifts of the higher order bits before the lower order bit shifts.  `3333222211110000`. Its easier to understand where the bits end up in the long.
The rest of the change is better focused.

Reply:
if change packDigits order, performance will be slow, I don't know why yet.

The following is the data running on MacBookPro M1 Max :
```
make test TEST="micro:java.util.UUIDBench.toString"

Benchmark           (size)   Mode  Cnt   Score   Error   Units (current order)
UUIDBench.toString   20000  thrpt   15  96.396 ? 0.946  ops/us


Benchmark           (size)   Mode  Cnt   Score   Error   Units (change packDigits order)
UUIDBench.toString   20000  thrpt   15  86.496 ? 0.542  ops/us
```

Reply:
Looks like something that might be an interesting puzzler for JIT compiler folks. Perhaps added implicit casts to long messes something up?